sdk-core/src/main/java/care/data4life/sdk/migration/RecordCompatibilityService.kt

```diff
@@ -19,138 +19,80 @@ package care.data4life.sdk.migration
-            .map { entry -> entry.key + TaggingContract.DELIMITER + tagHelper.normalize(entry.value) }
-            .let { normalizedTags ->
-                tagCryptoService.encryptList(
-                    normalizedTags,
-                    tagEncryptionKey
-                )
-            }
+    private fun mapTags(
+        tags: Tags,
+        tagEncryptionKey: GCKey
+    ): List<List<String>> {
+        return tags.map { tagGroup ->
+            Pair(
+                tagGroup.key + DELIMITER,
+                compatibilityEncoder.encode(tagGroup.value)
+            )
+        }.map { encodedTagGroup ->
+            encryptTags(
+                tagEncryptionKey,
+                encodedTagGroup.first,
+                encodedTagGroup.second
+            )
+        }
     }
 
-    @Throws(IOException::class)
-    private fun encryptAnnotations(
+    private fun mapAnnotations(
         annotations: Annotations,
         tagEncryptionKey: GCKey
-    ): MutableList<String> {
-        return tagCryptoService.encryptList(
-            annotations,
-            tagEncryptionKey,
-            ANNOTATION_KEY + TaggingContract.DELIMITER
-        )
-    }
-
-    private fun needsDoubleCall(
-        encodedAndEncryptedTags: List<String>,
-        encryptedTags: List<String>
-    ): Boolean = encodedAndEncryptedTags.sorted() != encryptedTags.sorted()
-
-    override fun searchRecords(
-        alias: String,
-        userId: String,
-        startDate: String?,
-        endDate: String?,
-        pageSize: Int,
-        offSet: Int,
-        tags: Tags,
-        annotations: Annotations
-    ): Observable<List<EncryptedRecord>> {
-        val (encodedAndEncryptedTags, encryptedTags) = encrypt(tags, annotations)
-        return if (needsDoubleCall(encodedAndEncryptedTags, encryptedTags)) {
-            Observable.zip(
-                apiService.searchRecords(
-                    alias,
-                    userId,
-                    startDate,
-                    endDate,
-                    pageSize,
-                    offSet,
-                    encodedAndEncryptedTags.joinToString(",")
-                ),
-                apiService.searchRecords(
-                    alias,
-                    userId,
-                    startDate,
-                    endDate,
-                    pageSize,
-                    offSet,
-                    encryptedTags.joinToString(",")
-                ),
-                BiFunction<List<EncryptedRecord>, List<EncryptedRecord>, List<EncryptedRecord>> { records1, records2 ->
-                    mutableListOf<EncryptedRecord>().also {
-                        it.addAll(records1)
-                        it.addAll(records2)
-                    }
-                }
+    ): List<List<String>> {
+        return annotations.map { annotation ->
+            Pair(
+                ANNOTATION_KEY + DELIMITER,
+                compatibilityEncoder.encode(annotation).copy(second = annotation)
```

Comment:
@alessioborraccino -> is this still issue?

Reply:
nope clear